_posts/2021-04-14-removing-implicit-coercions.markdown

```diff
@@ -0,0 +1,215 @@
+$x = 'abcd'; ++$x; // 'abce'
+```
+
+**Future Behaviour:**
+No operations will coerce. Additionally banning incrementing an empty string for consistency. The only valid values for inc/dec are ints, floats, and non-numeric strings. Note that while inc/dec a non-numeric string is still “legal,” it’s not necessarily a great idea.
+
+```
+$x = null; ++$x; // InvalidOperationException
+$x = ''; --$x; // InvalidOperationException
+$x = ''; ++$x; // InvalidOperationException
+$x = '1.123'; ++$x; // InvalidOperationException
+$x = '1bananas'; ++$x; // '1bananat'
+$x = 'abcd'; ++$x; // 'abce'
+```
+
+**Status of rollout:** 
+Soft Runtime Enforcement
+
+### Mathematical operations (`+`, `-`, `*`, `/`, `%`, `**`) and assignment equivalents
+
+**Current Behaviour:**
+null, bool, string, and resource will coerce to int / float depending on the other operand (defaulting to int for two non-num operands)
+
+```
+null + true; // 1
+```
+
+**Future Behaviour:**
+Operating on anything other than ints and floats will trigger an exception, with floats additionally being banned for use with `%`. It’s worth noting that we specifically decided to allow implicit coercions between ints and floats in this context.
+
+```
+null + true; // InvalidOperationException
+```
+
+**Status of rollout:** 
+Soft Runtime Enforcement
+
+### Comparison operations (`<` ,`<=`, `>` , `>=`, `<=>`)
+
+**Current Behaviour:**
+Attempting to list all the situations in which coercions happen and between which types would result in this post being twice as long. That should give you an idea of how much work this step will be.
+
+```
+null >= false; // true
+false < new Foo(); // true
+false < Foo::class; // true
+true < Foo::class; // false
+true > Foo::class; // false
+```
+
+**Future Behaviour:**
+Comparing values of two different types will throw an exception. A notable exception to this will be comparing ints and floats. In that case, comparison works as you’d expect, with the usual caveats about comparisons involving two approximately equal floats (including an approximately equal int and float).
+
+```
+null >= false; // InvalidOperationException
+false < new Foo(); // InvalidOperationException
+false < Foo::class; // InvalidOperationException
+true < Foo::class; // InvalidOperationException
+true > Foo::class; // InvalidOperationException
+```
+
+**Status of rollout:** 
+No Progress
+
+### Equality operations (`==`, `!=`)
+
+**Current Behaviour:**
+All the same combinations of types trigger coercions as with the comparison operators, but with *even more* situations resulting in coercions.
+
+```
+null == false; // true
+false == new Foo(); // false
+false == Foo::class; // false
+true == Foo::class; // true
+```
+
+**Future Behaviour:**
+There will be two steps to this, going through both Hard Runtime Enforcement and then New Behaviour. comparing values of two different types will throw an exception. Then, comparing values of two different types will simply be false. At the 
```

Comment:
The new sentence starts with "comparing" and might have had an abrupt end after "At the"

Reply:
yeah this is a really weird error